docs/spec/concepts.rst

```diff
@@ -3,6 +3,266 @@
 Type system concepts
 ====================
 
+Static, dynamic, and gradual typing
+-----------------------------------
+
+A **statically-typed** programming language runs a type-checker before running
+a program. The program is required to be well-typed according to the language's
+type system. The type system assigns a type to all expressions in the language
+and verifies that their uses obey the typing rules. Normally, a program that is
+not well-typed (i.e., one that contains a type error) will not run. Java and
+C++ are examples of statically-typed object-oriented languages.
+
+A **dynamically-typed** programming language does not run a type-checker before
+running a program. Instead, it checks the types of values before performing
+operations on them at runtime. This is not to say that the language is
+"untyped". Values at runtime have a type and their uses obey typing rules. Not
+every operation will be checked, but certain primitive operations in the
+language such as attribute access or arithmetic are. Python is a
+dynamically-typed language.
+
+**Gradual typing** is a way to combine static and dynamic typing.
+Type-annotated Python allows opting in to static type checking at a fine level
+of granularity, so that some type errors can be caught statically, before
+running the program. Variables, parameters, and returns can optionally be given
+a static type annotation. Even within the type of a single data structure,
+static type checking is opt-in and granular. For example, a dictionary can be
+annotated to have static checking of the key type but only dynamic checking of
+the value type.
+
+In gradual typing, a special "unknown" or "dynamic" type is used in the static
+type system for names or expressions whose type is not known statically.  In
+Python, this type is spelled :ref:`Any`. Because :ref:`Any` indicates a
+statically-unknown type, the static type checker can't check type correctness
+of operations on expressions typed as :ref:`Any`. These operations are
+dynamically checked, via the Python runtime's usual dynamic checking.
+
+The Python type system also uses ``...`` within :ref:`callable` and :ref:`Tuple
```

Comment:
Fixed. Thanks for the reminder that only `tuple[Any, ...]` is a gradual form. I updated the wording here to be a bit more careful so that it neither specifies this (that should be left for the tuple chapter) nor contradicts it.